[ContainerRegistry] Fix RegistryUri of ContainerRegistryImportSource

fengzhou-msft opened the pull request:
Resolve #31007
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

fengzhou-msft committed:
Fix RegistryUri of ContainerRegistryImportSource

Yao725 commented on sdk/containerregistry/Azure.ResourceManager.ContainerRegistry/src/Custom/Models/ContainerRegistryImportSource.Serialization.cs:
```diff
@@ -1,13 +1,12 @@
 
 using System.Text.Json;
 using Azure.Core;
 
```
I don't prefer to totally remove this model to the customized code, it might let us miss some new properties if the service team update this model in the future. How about we rename the property to `RegistryAddress` and write some code as shown below in the customized code:
```csharp
[Obsolete("RegistryUri is deprecated, use RegistryAddress instead")]
[EditorBrowsable(EditorBrowsableState.Never)]
public Uri RegistryUri {
    get
    {
        if (RegistryAddress != null && Uri.TryCreate(RegistryAddress, UriKind.RelativeOrAbsolute, out var registryUri))
            return registryUri;
        return null;
    }
    set
    {
        RegistryAddress = value == null ? null : value.ToString();
    }
}
```

fengzhou-msft replied:
This seems a good workaround to avoid customizing the `write` method in serialization class (and the generator does not support only customize `write` method, that's why I need to customize the whole class).

fengzhou-msft replied:
Relative Uri is discouraged with https://github.com/Azure/azure-sdk-for-net/blob/main/eng/BannedSymbols.txt, will modify to only use AbsoluteUri.

fengzhou-msft committed:
address comment

Yao725 reviewed (approved):
LGTM

azure-sdk committed:
Increment version for containerregistry releases (#31936)

Increment package version after release of Azure.ResourceManager.ContainerRegistry

fengzhou-msft committed:
merge main

fengzhou-msft committed:
update changelog

azure-sdk commented:
**API change check**

APIView has identified API level changes in this PR and created following API reviews.

[Azure.ResourceManager.ContainerRegistry](https://apiview.dev/Assemblies/Review/46a62a5286164c2a89eb38b219336f4f)
